### ISSUE TITLE
Current trunk doesn't build on ocaml 4.12.0

### ISSUE THREAD

**Issue by anuragsoni** [janestreet/core_kernel#105]
It looks like this patch https://github.com/janestreet/core_kernel/commit/7e507516e02985898b258e187f52476d537b6e04#diff-8ecc04e8fa8ca2e42aa9c3be5f580180571d3b3e5e33ccff52b4937891ad14d6 
from 0.14.1 is missing in the current trunk. At the moment there is a build error when trying to build core_kernel on ocaml 4.12.0

```
File "core/src/gc.ml", lines 19-37, characters 4-52:
19 | ....type t = Caml.Gc.stat =
20 |       { minor_words : float
21 |       ; promoted_words : float
22 |       ; major_words : float
23 |       ; minor_collections : int
...
34 |       ; top_heap_words : int
35 |       ; stack_size : int
36 |       }
37 |     [@@deriving compare, hash, bin_io, sexp, fields]
Error: This variant or record definition does not match that of type
         Caml.Gc.stat
       The field forced_major_collections is only present in the original definition.
```

**Comment by aalekseyev** [comment]
Should be fixed in 8b456197af943e84d49c5eec1ee0abe1e6da7b96.

**Comment by anuragsoni** [comment]
@aalekseyev Thank you!
